pkg/asset/installconfig/gcp/validation.go

```diff
@@ -0,0 +1,74 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	compute "google.golang.org/api/compute/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Validate executes platform-specific validation.
+func Validate(client API, ic *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, validateNetworks(client, ic, field.NewPath("platform").Child("gcp"))...)
+
+	return allErrs.ToAggregate()
+}
+
+// validateNetworks checks that the user-provided VPC is in the project and the provided subnets are valid.
+func validateNetworks(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.GCP.Network != "" {
+		_, err := client.GetNetwork(context.TODO(), ic.GCP.Network, ic.GCP.ProjectID)
+		if err != nil {
+			return append(allErrs, field.Invalid(fieldPath.Child("network"), ic.GCP.Network, err.Error()))
+		}
+
+		subnets, err := client.GetSubnetworks(context.TODO(), ic.GCP.Network, ic.GCP.ProjectID, ic.GCP.Region)
+		if err != nil {
+			return append(allErrs, field.Invalid(fieldPath.Child("network"), ic.GCP.Network, "failed to retrieve subnets"))
+		}
+
+		allErrs = append(allErrs, validateSubnet(client, ic, fieldPath, subnets, "computeSubnet", ic.GCP.ComputeSubnet)...)
```

Comment:
we should pass the `fieldPath.Child("computeSubnet")` here rather than passing both the fieldpath and field name.